### ISSUE TITLE
<p:tooltip> does not disappear after another Ajax request

### ISSUE THREAD

**Issue by artemyankovich** [primefaces/primefaces#1014]
Hi!
I found bug in tooltip work. For example, i have these components in my view:

```
            <p:selectCheckboxMenu id="test_checkbox_menu" label="Test" onHide="testRC()">
        <f:selectItem itemLabel="One" />
    </p:selectCheckboxMenu>     

    <p:remoteCommand name="testRC" update="test_checkbox_menu" />

    <p:tooltip for="test_checkbox_menu">
        <h:outputText value="Test" />
    </p:tooltip>        
```

When client side callback execute when hide selectCheckboxMenu and in this moment tooltip is visible, this tooltip freeze on page and never dissapeared until full page reload. This occurs because after update selectCheckboxMenu by remoteCommand appers new div element in html page, same as tooltip div(even same id), but with css style display:block. Initial tooltip div looks ok and have display:none. As a result we have normal tooltip div and strange div with display:block css style, but only one widget class for initial tooltip. New strange div element will appers after each onHide calls.  

Bug manifests itself not only for this case, but in all cases where tooltip for component is visible and component rerendered in this moment.(For IE9 in all cases, for Chrome i found only with selectCheckboxMenu)

Also, I found post about this bug in StackOverflow http://stackoverflow.com/questions/31428751/primefaces-ptooltip-does-not-disappear-after-another-ajax-request

**Comment by lcsdms** [comment]
I'm having the same problem with p:panel and p:inputText

`<p:panel id="test_panel">
                            <p:inputText id="test_input"> 
                                <p:ajax event="change" update="test_panel"/>
                            </p:inputText>
                            <p:tooltip id="tooltip_id" for="test_input" showEvent="focus" hideEvent="blur" styleClass="tp-com-imagem"   >
                                    <h:graphicImage value="/resources/imagens/etiqueta-paciente.png"/>
                                    Some Text<br></br>
                                    Some Text<br></br>
                                    Some text                   
                            </p:tooltip>
                        </p:panel>`

Checked and it creates a new div with the same tooltip id, with display:block, I needed to use this workaround to remove the nodes, using oncomplete on the ajax call:

```
       `function hideTooltips() {

            var element = document.getElementById("tooltip_id");

            while(element.style.display != "block"){
                element = element.nextSibling;             
            }
            element.parentNode.removeChild(element);


        }`
```

It happens on Chrome, but on IE11 I don't have that problem.

**Comment by mertsincan** [comment]
I'll review and get back to you.

**Comment by husnutapan** [comment]
you can try both together itemLabel and itemValue

**Comment by cnsgithub** [comment]
Hi, Still happens on PF 6.0 Final (and current ELITE v.) 

To reproduce, Click on something which does an ajax call, then quickly move the mouse to the tooltip before dom is updated, then the tooltip will persist, even if it is no longer in dom. 

Update:
Another jQuery based workaround:

```
<p:ajaxStatus oncomplete="cleanupTooltips();" />
```

```
function cleanupTooltips(){
    $('.ui-tooltip').each(function(index, elem){
        $(elem).hide();
    });
}
```

**Comment by tandraschko** [comment]
Updating the tooltip works fine for me to remove it.

**Comment by jheidlage1222** [comment]
So I'm dealing with a similar issue, I have an ajax request being triggered by a commandButton and the tooltip doesn't hide after clicking it and moving away.  At first I used @cnsgithub s solution which worked quite well.  I was about to put a pin in this and move on but I found something.  I can set the hideEvent attrib to two events, one for your standard mouseout but the second for when I click my button and I just tested in both IE and Chrome and it works in both cases.  
`<p:tooltip id="saveDeleteTooltip" globalSelector="[data-tooltip]" position="top" hideEvent="mouseleave click"/>`
